controllers/openstackmachine_controller.go

```diff
@@ -145,6 +145,12 @@ func (r *OpenStackMachineReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return reconcile.Result{}, err
 	}
 
+	// Resolve and store referenced resources
+	err = compute.ResolveReferencedMachineResources(scope, &openStackMachine.Spec, &openStackMachine.Status.ReferencedResources)
```

Comment:
Could you please explain why we need this? I'm sure this was explained somewhere but I can't find it in the review. Why don't we just resolve it in `machineToInstanceSpec` directly or somewhere around these lines?

I guess my questions are:
* Why so early in the reconcile, and why not in `machineToInstanceSpec` or somewhere else around it
* Are we creating a new pattern with this "resolved referenced machine resources"? If yes we need to document this.

Reply:
Okay, yeah, this is a new pattern. Documenting it is a good idea, I'll try to find a place to do it. Anyway doing it here gives us one advantage - it declutters `machineToInstanceSpec()` and we call `machineToInstanceSpec()` in the delete flow too. If we'd put all the resources related to the machine in the status, we should be able to make the delete flow simpler too.

Reply:
I like the idea. :+1:

Reply:
The only thing that bothers me is that we call the compute package, while in the future I suspect we'll have networking & storage resources. Could we please find a common place.